### ISSUE TITLE
[@types/react-redux] connect without mergeProps as third argument

### ISSUE THREAD

**Issue by mtoepfl** [DefinitelyTyped/DefinitelyTyped#48107]
Is something wrong with CI or publishing infrastructure?

- Check the [Infrastructure status updates](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/44317) thread for known issues.
- Leave a message on the [DefinitelyTyped Gitter](http://gitter.im/DefinitelyTyped/DefinitelyTyped).

If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @tkqubo @kenzierocks @clayne11 @tansongyang @nicholasboll @mdibyo @kallikrein @val1984 @jrakotoharisoa @apapirovski @surgeboris @soerenbf @mrwolfz @dylanvann @Lazyuki @kazuma1989 

If you do not mention the authors the issue will be ignored.

Thank you, for all the work you did already on @types/react-redux making our lifes easier!

I would like to suggest an improvement:

```typescript
export default connect<
  TStateProps,
  TDispatchProps,
  TOwnProps,
  TProps,
  StateRoot
>(
  mapStateToProps,
  mapDispatchToProps,
)(App);
```
gives this error

``` typescript
(alias) connect<TStateProps, TDispatchProps, TOwnProps>(mapStateToProps: null | undefined, mapDispatchToProps: MapDispatchToPropsNonObject<TDispatchProps, TOwnProps>): InferableComponentEnhancerWithProps<...> (+14 overloads)
import connect
Expected 3-4 arguments, but got 2.ts(2554)
index.d.ts(235, 9): An argument for 'mergeProps' was not provided.
```

**Would be easier for us/everybody, to leave the TProps argument despite there is no mergeProps.**

Solution:

change overload for type: 

``` typescript
    <TStateProps = {}, TDispatchProps = {}, TOwnProps = {}, TMergedProps = {}, State = DefaultRootState>(
        mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps, State>,
        mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>,
        mergeProps: MergeProps<TStateProps, TDispatchProps, TOwnProps, TMergedProps>,
        options?: Options<State, TStateProps, TOwnProps, TMergedProps>
    ): InferableComponentEnhancerWithProps<TMergedProps, TOwnProps>;
```
to 

``` typescript
    <TStateProps = {}, TDispatchProps = {}, TOwnProps = {}, TMergedProps = {}, State = DefaultRootState>(
        mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps, State>,
        mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>,
        mergeProps?: MergeProps<TStateProps, TDispatchProps, TOwnProps, TMergedProps> | null | undefined,
        options?: Options<State, TStateProps, TOwnProps, TMergedProps>
    ): InferableComponentEnhancerWithProps<TMergedProps, TOwnProps>;
```

in short add

**mergeProps?** and  **| null | undefined**

Risks:
no, as far as i can see

Thank you

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
